src/DMLQueryBuilder.php

```diff
@@ -74,42 +69,28 @@ public function upsert(
     ): string {
         $insertSql = $this->insert($table, $insertColumns, $params);
 
-        /** @psalm-var array $uniqueNames */
-        [$uniqueNames, , $updateNames] = $this->prepareUpsertColumns(
-            $table,
-            $insertColumns,
-            $updateColumns,
-        );
+        [$uniqueNames, , $updateNames] = $this->prepareUpsertColumns($table, $insertColumns, $updateColumns);
 
         if (empty($uniqueNames)) {
             return $insertSql;
         }
 
-        if ($updateNames === []) {
+        if ($updateColumns === false || $updateNames === []) {
             /** there are no columns to update */
-            $updateColumns = false;
-        }
-
-        if ($updateColumns === false) {
             return "$insertSql ON CONFLICT DO NOTHING";
         }
 
         if ($updateColumns === true) {
             $updateColumns = [];
 
-            /** @psalm-var string $name */
+            /** @psalm-var string[] $updateNames */
```

Comment:
```suggestion
```

Seems, it not need after last changes in PR to yiisoft/db.

Reply:
`$updateNames` could be `null` but if `$updateColumns === true` it is `array`

https://github.com/yiisoft/db/pull/746/files#diff-697d8c4d18282e94c125fbb9302841ce34c6f6bbe6a4406cd57b371ca9e145aeR299-R303